tests/integration/local/invoke/invoke_cdk_templates.py

```diff
@@ -0,0 +1,59 @@
+import json
+import docker
+from docker.errors import APIError
+from parameterized import parameterized
+
+from samcli import __version__ as version
+from samcli.local.docker.lambda_image import RAPID_IMAGE_TAG_PREFIX
+from tests.integration.local.invoke.invoke_integ_base import InvokeIntegBase
+from samcli.lib.utils.architecture import X86_64
+
+
+class TestCDKSynthesizedTemplatesFunctions(InvokeIntegBase):
+
+    template = "cdk/cdk_template.yaml"
+    functions = [
+        "StandardFunctionConstructZipFunction",
+        "StandardFunctionConstructImageFunction",
+        "DockerImageFunctionConstruct",
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        # Run sam build first to build the image functions
+        super(TestCDKSynthesizedTemplatesFunctions, cls).setUpClass()
+        build_command_list = super().get_build_command_list(cls, template_path=cls.template_path)
+        super().run_command(cls, command_list=build_command_list)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        # Remove images
+        docker_client = docker.from_env()
+        for function in cls.functions:
+            try:
+                docker_client.api.remove_image(f"{function.lower()}")
+                docker_client.api.remove_image(f"{function.lower()}:{RAPID_IMAGE_TAG_PREFIX}-{version}-{X86_64}")
+            except APIError:
+                pass
+
+    @parameterized.expand(functions)
+    def test_build_and_invoke_image_function(self, function_name):
```

Comment:
Most of our docker tests are re-runs due to weird networking or docker issues: https://github.com/aws/aws-sam-cli/blob/develop/tests/integration/local/invoke/test_integrations_cli.py#L35 

Should we do the same here?

Reply:
Looks like we also skip build tests on Windows in CI. Not exactly sure why but I'm assuming some Docker weirdness. Will add that in too.

Reply:
We skip build tests because we run them separately: https://github.com/aws/aws-sam-cli/blob/develop/appveyor-windows-build-dotnet.yml

Might make sense to add these to the correct language windows appveyor job.